site/src/components/Seo.tsx

```diff
@@ -0,0 +1,81 @@
+import { graphql, useStaticQuery } from "gatsby";
+import React from "react";
+
+export function Seo ({
+  title,
+  description,
+  children
+} : {
+  title?: string,
+  description?: string,
+  children?: React.ReactNode
+}) {
+  const data = useStaticQuery(graphql`
+    query {
+      site {
+        siteMetadata {
+          title
+          host
+          tagline
+          description
+          author
+        }
+      }
+    }
+  `)
+  const metaTitle = `${data.site.siteMetadata.title} 〜 ${data.site.siteMetadata.tagline} 〜`
+  const customTitle = title ? `${title} | ${metaTitle}` : metaTitle
+
+  const metaDescription = data.site.siteMetadata.description
+  const customDescription = description ? description : metaDescription
```

Comment:
titleと同じ理屈でデフォルトもしくはページごとのdescriptionを返します。
titleがページごとに変わるなら、こちらも変えたほうが良いと考え追加しました。


### top
![スクリーンショット 2024-11-19 17 30 54のコピー](https://github.com/user-attachments/assets/edbcb653-0d2c-4b3c-835f-dac685ad3593)


### about
![スクリーンショット 2024-11-19 17 31 06のコピー](https://github.com/user-attachments/assets/b2a16c1b-45a0-4c5e-9343-59712f3b8395)


### terms_of_use
![スクリーンショット 2024-11-19 17 30 38のコピー](https://github.com/user-attachments/assets/72afddf1-ea4d-4d58-bcba-678ece3cb3b2)

Reply:
@okky-hi OKと思います。以下のように書けそうな気がするので一度試してみてください。

```suggestion
  const metaDescription = data.site.siteMetadata.description
  const customDescription = description || metaDescription
```